### PR TITLE
feat(docs): add wsl setup to docs

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

This PR adds the WSL setup for the freeCodeCamp local setup using Windows

### CODE REVIEW

**Comment by Sky020** [comment]
@raisedadead , Before I remove draft status on this, is there anything you would want done/redone?

**Comment by raisedadead** [comment]
Hi @Sky020 

Thanks for the PR. I sincerely thank you for taking the time to come up with this draft. Please bear with us while we validate this.

I would recommend reading through the tone of the other guides that we have had and make it more netural in tone? I would also recommend eyes from others for a quick look.

**Comment by ojeytonwilliams** [comment]
@Sky020 I'd be happy to proofread this.  Feel free to ping me once you're done with Mrugesh's request.

**Comment by Sky020** [comment]
@ojeytonwilliams & @raisedadead , I would appreciate your input for any and all changes to this PR.

File: docs/how-to-setup-wsl.md
Comment (Sky020) [inline docs/how-to-setup-wsl.md:80]:
```suggestion
> Provided you have not deviated from the default setup, the `IdentityFile` will be as above.
```

File: docs/how-to-setup-wsl.md
Comment (raisedadead) [inline docs/how-to-setup-wsl.md:3]: It would be a nice to add some requirements of WSL itself. I might be mistaken, but it requires you to be on Windows 10 Pro and I think WSL2 needs a may 2020 update.

File: docs/how-to-setup-wsl.md
Comment (raisedadead) [inline docs/how-to-setup-wsl.md:23]: This suggests that any of the alternative is valid and should work, but it looks like they need to try them in an order?

File: docs/how-to-setup-wsl.md
Comment (raisedadead) [inline docs/how-to-setup-wsl.md:84]: If we are asking users to do the steps in this part, we are expecting a lot from them. Meaning, we are maintainers have gotten this wrong and need to provide a better way.

**Comment by raisedadead** [comment]
After going thru this, I am of the opinion that we should invest and investigate native support for developing on Windows 10 (minus WSL).

**Comment by Sky020** [comment]
I will admit, all of the issues are aside from the freeCodeCamp code base. So far, many of the errors I have encountered have been to do with Linux based issues with Git, and the lack of native support for MongoDB on WSL.

There is no need to include this in the docs, and I can understand the fact that this is not a fCC doc.

Personally, I am glad I have gone through the setup of Windows, WSL, and Ubuntu (on a separate laptop), and perhaps this doc can be refactored into a guide post?

**Comment by ojeytonwilliams** [comment]
> investigate native support for developing on Windows 10 

Shaun's gone through this more recently, so he's better placed to comment, but Windows 10 is pretty bad for node apps.  If memory serves the installation took roughly an hour on my machine.  I don't understand why, but it seems that Windows has difficulty with the sheer number of files in a typical node install.

My intuition is that the best way to support Windows contributors is to try and fix any problems with Docker and improve that guide.

I had hoped WSL would be the answer, but probably not unless we can cut the number of steps.

**Comment by raisedadead** [comment]
> There is no need to include this in the docs, and I can understand the fact that this is not a fCC doc.

I think the very fact that you had to go thru this makes it a requirement to include it as an ancillary doc.

> I had hoped WSL would be the answer, but probably not unless we can cut the number of steps.

The bigger question for us to address the native support. 

Windows 10 as a Node dev environment has come miles than it used to be, with recent inclusion of `winget` CLI (Windows's answer to Homebrew, apt-get) and Windows Terminal (with Powershell).

Let me circle around to this over the weekend on my old laptop.

File: docs/how-to-setup-wsl.md
Comment (Sky020) [reply docs/how-to-setup-wsl.md:3]: I would be fine doing this, but I left it out, seeing as Microsoft take users through the requirements on the WSL Docs page. Should I just add a small note anyway?

File: docs/how-to-setup-wsl.md
Comment (Sky020) [reply docs/how-to-setup-wsl.md:23]: I will reword this. I suppose their is an order of hopefulness. It is not structured in an order I followed, but I just come into a lot of errors.

The alternatives are there as a hopeful way to fix any issues pertaining to Git. The order is based on length.

File: docs/how-to-setup-wsl.md
Comment (raisedadead) [reply docs/how-to-setup-wsl.md:3]: Yeah, that would be useful because users can avoid going through all the caveats that Microsoft has with the 'Pro' version of OS and the need for being on the latest update.

File: docs/how-to-setup-wsl.md
Comment (raisedadead) [inline docs/how-to-setup-wsl.md:8]: > `(As of June, only available once part of the Windows Insider Program)`

Can you confirm this is correct? I believe its available globally. I think I was able to install it on Windows 10 normally without having signed up for the insider program.

Comment (Sky020) [reply docs/how-to-setup-wsl.md:8]: Right, it should be the other way around. The version was released on the 27/05

Comment (raisedadead) [reply docs/how-to-setup-wsl.md:8]: Did you get a chance to remove the line? I think this is good to go pending the change.

File: docs/how-to-setup-wsl.md
Comment (ojeytonwilliams) [inline docs/how-to-setup-wsl.md:12]:
```suggestion
This guide covers some common issues encountered with the setup of WSL.
```

File: docs/how-to-setup-wsl.md
Comment (ojeytonwilliams) [inline docs/how-to-setup-wsl.md:34]:
```suggestion
1) Update your distribution:
```

File: docs/how-to-setup-wsl.md
Comment (ojeytonwilliams) [inline docs/how-to-setup-wsl.md:60]:
```suggestion
3) `cd` into the `.ssh` folder, and create a config file:
```

**Review by ojeytonwilliams — DISMISSED** [review]
Nitpicks:

**Comment by Sky020** [comment]
@raisedadead , Ja, got removed the inaccurate requirements.
Thank you, @ojeytonwilliams , for catching those typos.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**
